plugins/@grouparoo/pardot/src/lib/export/exportProfiles.ts

```diff
@@ -0,0 +1,273 @@
+import {
+  buildBatchExports,
+  BatchExport,
+  exportProfilesInBatch,
+  BatchSyncMode,
+  BatchGroupMode,
+  BatchMethodGetClient,
+  BatchMethodFindAndSetDestinationIds,
+  BatchMethodDeleteByDestinationIds,
+  BatchMethodUpdateByDestinationIds,
+  BatchMethodCreateByForeignKeyAndSetDestinationIds,
+  BatchMethodAddToGroups,
+  BatchMethodNormalizeForeignKeyValue,
+  BatchMethodNormalizeGroupName,
+  BatchMethodRemoveFromGroups,
+} from "@grouparoo/app-templates/dist/destination/batch";
+import { ExportProfilesPluginMethod } from "@grouparoo/core";
+import PardotClient from "../client";
+import { connect } from "../connect";
+import { PardotCacheData, getListId } from "./listMethods";
+
+interface PardotData {
+  cacheData: PardotCacheData;
+}
+
+// return an object that you can connect with
+const getClient: BatchMethodGetClient = async ({ config }) => {
+  return connect(config.appOptions);
+};
+
+// fetch using the keys to set destinationId and result on BatchExports
+// use the getByForeignKey to lookup results
+const findAndSetDestinationIds: BatchMethodFindAndSetDestinationIds = async ({
+  client,
+  foreignKeys,
+  getByForeignKey,
+}) => {
+  for (const email of foreignKeys) {
+    const prospect = await client.getProspectByEmail(email);
+    if (prospect) {
+      const user = getByForeignKey(email);
+      user.destinationId = prospect.id;
+      user.result = prospect;
+    }
+  }
+};
+
+// delete the given destinationIds
+const deleteByDestinationIds: BatchMethodDeleteByDestinationIds = async ({
+  client,
+  users,
+}) => {
+  for (const user of users) {
+    // No batch delete available, do one by one
+    await client.deleteProspectById(parseInt(user.destinationId));
```

Comment:
it's likely they are there but if they aren't here, maybe we can catch. timing issues and it's fine as long as they are gone.

Reply:
Not entirely sure what you're suggesting to do on error. Wrap in try/catch and set error to the specific user and let grouparoo retry it?

Reply:
```
try {
  await client.deleteProspectById(parseInt(user.destinationId));
} catch(error) {
  user.error = error;
}
```

Reply:
Then Grouparoo will let the other ones work and try this one again later.